### PR TITLE
test for installing from remote file

### PR DESCRIPTION
**Pull request by Dreamersoul**
i am opening this pull request even though i didn't have a chance to run the test today (windows tests are broken i think #882) will take a look using my mac tomorrow
in the mean time please let me know if i used the correct decorators for the test function and if i am missing something from the test case

thank you for the amazing work

this is  a test for: `pipenv install -r http://something.com/req.txt`

### CODE REVIEW

File: tests/test_pipenv.py [hunk 1]
```diff
@@ -796,3 +796,21 @@ def test_local_zipfiles(self):
...
+    @pytest.mark.install
+    @pytest.mark.files
+    @pytest.mark.urls
+    def test_install_remote_requirments(self):
+        with PipenvInstance() as p:
+            # using a github hosted requirements.txt file
+            c = p.pipenv('install -r https://github.com/kennethreitz/pipenv/blob/3688148ac7cfecefb085c474b092c31d791952c1/tests/test_artifacts/requirements.txt')
+            assert c.return_code == 0
+
+            # should i check versions here? or there is another test for that? hmm
```
Comment (nateprewitt) [inline tests/test_pipenv.py]: It wouldn't hurt to check versions.

File: tests/test_pipenv.py [hunk 2]
```diff
@@ -796,3 +796,21 @@ def test_local_zipfiles(self):
...
             assert 'file' in dep or 'path' in dep
+
+
+    @pytest.mark.install
+    @pytest.mark.files
+    @pytest.mark.urls
+    def test_install_remote_requirments(self):
+        with PipenvInstance() as p:
+            # using a github hosted requirements.txt file
+            c = p.pipenv('install -r https://github.com/kennethreitz/pipenv/blob/3688148ac7cfecefb085c474b092c31d791952c1/tests/test_artifacts/requirements.txt')
```
Comment (nateprewitt) [inline tests/test_pipenv.py]: This is going to need to be `https://raw.githubusercontent.com/kennethreitz/pipenv/3688148ac7cfecefb085c474b092c31d791952c1/tests/test_artifacts/requirements.txt` actually. The link I originally provided is a full HTML page from github, not the raw file.

**Comment by Dreamersoul** [comment]
Thanks for the amazing input will look into it today

**Comment by Dreamersoul** [comment]
commits are squashed and test was passing locally. we should be all set 👍

File: tests/test_pipenv.py [hunk 3]
```diff
@@ -796,3 +796,23 @@ def test_local_zipfiles(self):
             dep = p.lockfile['default'][key]
 
             assert 'file' in dep or 'path' in dep
+
```
Comment (erinxocon) [inline tests/test_pipenv.py]: This test fails on windows.  We need it to work before merging.

Comment (erinxocon) [inline tests/test_pipenv.py]: This test fails on windows. We need it to work before merging.

Comment (erinxocon) [reply tests/test_pipenv.py]: This might actually be a bug on windows in pipenv due to how windows opens and places files locks on things.

Comment (Dreamersoul) [reply tests/test_pipenv.py]: could this be the problem?
https://bugs.python.org/msg27445

it says the file is probably locked by the process
we could probably unlock the file or something?

**Comment by Dreamersoul** [comment]
so after investigation turns out `tempfile.mkstemp` actually opens the file and returns the file descriptor so we have two solutions:
1. close the file immediately so basically using mkstemp just to create the file
2. we could hold on to the descriptor and close it before deleting

which one do you recommend? also if i got it wrong can someone suggest another approach

i have already implemented the second approach do you want me to push it?

**Comment by techalchemy** [comment]
Oh this test... it’s probably more likely that it’s failing due to issues relating to loca file installation. I’m in the middle of untangling how that wound up broken

**Comment by techalchemy** [comment]
Ah this is for the other test. I guess my question would be whether using a descriptor is the best option at all. We already create temp files for requirements as we resolve and add to Pipfiles. Is there any reason not to just iterate over requirements from the remotesfile contents? I haven’t looked at complexity for that but you’re not streaming the file content into the descriptor right? So it’s going to be all loaded in memory anyway

**Comment by Dreamersoul** [comment]
@techalchemy I was thinking of streaming it but went with using temp files!
I wasn't using file descriptors but the tempfile library automatically opens the file

I'm just downloading the file you can check at the end of the utils.py file to see how I am doing it :)

**Comment by nateprewitt** [comment]
I've pushed b1be8f7 which will properly capture and then close the file descriptor for Windows before deleting the temp file.

If you rebase onto master @Dreamersoul, I think we're about ready.

**Comment by Dreamersoul** [comment]
Awesome! I checked your commit and turns out I did the same locally! Once I have access to my laptop I'll rebase and push :+1:

**Comment by Dreamersoul** [comment]
Can someone trigger a rebuild? The tests failed because of pip and not my test :sweat_smile:

**Comment by nateprewitt** [comment]
Looks like we're set. Thanks again @DreamerSoul!

File: tests/test_pipenv.py [hunk 4]
```diff
@@ -804,3 +804,23 @@ def test_local_zipfiles(self):
...
+    @pytest.mark.install
+    @pytest.mark.files
+    @pytest.mark.urls
+    def test_install_remote_requirments(self):
+        with PipenvInstance() as p:
+            # using a github hosted requirements.txt file
+            c = p.pipenv('install -r https://raw.githubusercontent.com/kennethreitz/pipenv/3688148ac7cfecefb085c474b092c31d791952c1/tests/test_artifacts/requirements.txt')
+            assert c.return_code == 0
+
+            print(p.lockfile['default'])
```
Comment (nateprewitt) [inline tests/test_pipenv.py]: Ah, actually can you remove this print?

Comment (Dreamersoul) [reply tests/test_pipenv.py]: I thought I did *face palm*

**Comment by nateprewitt** [comment]
✨

**Pull request merged**